src/main/java/com/ibm/stocator/fs/swift/SwiftAPIClient.java

```diff
@@ -301,34 +309,23 @@ public FileStatus getObjectMetadata(String hostName,
       1) a zero byte object with the name of the directory
       2) no zero byte object with the name of the directory
     */
-    String objectName = path.toString().substring(hostName.length());
-    String objectNameNoSlash = objectName;
-    if (objectName.endsWith("/")) {
-      /*
-        removing the trailing slash because it is not supported in Swift
-        an request on an object (not a container) that has a trailing slash will lead
-        to a 404 response message
-      */
-      objectNameNoSlash = objectName.substring(0, objectName.length() - 1);
-    }
-    StoredObject so = cont.getObject(objectNameNoSlash);
     boolean isDirectory = false;
-    if (so.exists()) {
-      // We need to check if the object size is equal to zero
+    String objectName = getObjName(hostName, path);
+    Long objContentLength = cachedStoredObjectSize.get(objectName);
+    if (objContentLength != null) {
+      // object exists, We need to check if the object size is equal to zero
       // If so, it might be a directory
-      long contentLength = so.getContentLength();
-      String lastModified = so.getLastModified();
-      if (contentLength == 0) {
+      if (objContentLength == 0) {
         Collection<DirectoryOrObject> directoryFiles = cont.listDirectory(objectName, '/', "", 10);
         if (directoryFiles != null && directoryFiles.size() != 0) {
           // The zero length object is a directory
           isDirectory = true;
         }
       }
       LOG.trace("{} is object. isDirectory: {}  lastModified: {}", path.toString(),
-          isDirectory, lastModified);
-      return new FileStatus(contentLength, isDirectory, 1, blockSize,
-              getLastModified(lastModified), path);
+          isDirectory, 0);
+      return new FileStatus(objContentLength, isDirectory, 1, blockSize,
+             0L, path);
```

Comment:
@guyGerson this code is not used by Spark only, but can be used by other Hadoop eco system. 
Can you address the rest of the comments please and last modified to cache? We need this ASAP. 
Thanks